parser.js

```diff
@@ -20,6 +20,10 @@ const visitorKeys = require("./visitor-keys");
 exports.version = require("./package.json").version;
 
 exports.parseForESLint = function parseForESLint(code, options) {
+    if (options && typeof options.filePath === "string" && options.filePath.endsWith(".tsx")) {
```

Comment:
I wonder if the behavior should be that it only auto-detects when the `jsx` option is not defined? I think it could be confusing for a user to have `jsx: false` and still have it parse JSX.

Reply:
Hm. After I re-thought it:

On TypeScript, we cannot enable jsx on `*.ts` files and cannot disable jsx on `*.tsx` files.
So probably makes sense if it uses the `jsx` option only when filename is not provided.

Reply:
Ah... No, when neither `*.ts`  or `*.tsx` then it use the given `jsx` option.